### PR TITLE
Filter flags obtained by pc files

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: Sources/PkgConfig/whitelist().swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                filtered += [nextFlag] 
+            }
+        }
+        return filtered
+    }
+
+    let cFlagsFiltered = filter(flags: flags.cFlags, filters: [includePath])
+    let libsFiltered = filter(flags: flags.libs, filters: [linkPath, linkFlag])
+    return (cFlagsFiltered, libsFiltered) 
+}
```
Comment (ddunbar) [inline Sources/PkgConfig/whitelist().swift]: I don't see a reason to put this in its own file...

**Comment by ddunbar** [comment]
I don't think we should filter the flags without throwing errors on other flags. Otherwise I think we risk just exposing the user to weird behavior that doesn't match either what the package wanted, or what our system knows how to support.

**Comment by aciidb0mb3r** [comment]
should we go ahead with the build and print out warnings that these flags are not going to be included or error out completely?

**Comment by ddunbar** [comment]
I would prefer to error out. The general benefit to an error is that we then know for sure that no one is relying on the behavior.

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
This passed but got conflicted. Tested locally. @ddunbar review please.

**Comment by aciidb0mb3r** [comment]
huh weird OSX CI got triggered somehow

**Comment by aciidb0mb3r** [comment]
might as well run linux too then 
@swift-ci please test linux

**Comment by aciidb0mb3r** [comment]
please test linux

**Comment by aciidb0mb3r** [comment]
@swift-ci please test linux

File: Sources/PkgConfig/Module+PkgConfig.swift [hunk 2]
```diff
@@ -82,3 +83,40 @@ private extension SystemPackageProvider {
...
 }
+
+private struct FlagsFilter {
+    let flag: String
+    let hasValue: Bool
+}
+
+private let includePath = FlagsFilter(flag: "-I", hasValue: true)
+private let linkPath = FlagsFilter(flag: "-L", hasValue: true)
+private let linkFlag = FlagsFilter(flag: "-l", hasValue: false)
```
Comment (ddunbar) [inline Sources/PkgConfig/Module+PkgConfig.swift]: This isn't correct, actually the compiler driver will accept all forms of these arguments with and without a space between the value. This should make the code simpler.

Also, we should probably accept `-F` (framework search path) and `-framework` even though they probably aren't used much.

Comment (aciidb0mb3r) [inline Sources/PkgConfig/Module+PkgConfig.swift]: ah.. Done

**Comment by aciidb0mb3r** [comment]
rebased

**Pull request merged**